rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -0,0 +1,484 @@
+mod client;
+
+pub use client::Client;
+
+// // OPEN QUESTIONS
+// TODO Except connector this client is used somewhere else ?
+// TODO Do we need to validate any required fields ? If so most probbaly need a builder
+
+/// Represents an error that occurred in the Azure Device Registry Client implementation.
+#[derive(Debug, Error)]
+#[error(transparent)]
+pub struct AzureDeviceRegistryError(#[from] ErrorKind);
+
+impl AzureDeviceRegistryError {
+    /// Returns the [`ErrorKind`] of the error.
+    #[must_use]
+    pub fn kind(&self) -> &ErrorKind {
+        &self.0
+    }
+}
+
+/// Represents the kinds of errors that occur in the Azure Device Registry implementation.
+#[derive(Error, Debug)]
+#[allow(clippy::large_enum_variant)]
+pub enum ErrorKind {
+    /// An error occurred in the AIO Protocol. See [`AIOProtocolError`] for more information.
+    #[error(transparent)]
+    AIOProtocolError(AIOProtocolError),
+    /// An error occurred during serialization of a request.
+    #[error("{0}")]
+    SerializationError(String),
+    /// An argument provided for a request was invalid.
+    #[error("{0}")]
+    InvalidArgument(String),
+    /// An error was returned by the Azure Device Registry Service.
+    #[error("{0:?}")]
+    ServiceError(ServiceError),
+}
+
+/// An error returned by the Azure Device Registry Service.
+/// // TODO Ask service team about what tsrcuture of service errors ? And redefine the struct based on that.
+#[derive(Debug)]
+pub struct ServiceError {
+    /// The error message.
+    pub message: String,
+    /// The name of the property associated with the error, if present.
+    pub property_name: Option<String>,
+    /// The value of the property associated with the error, if present.
+    pub property_value: Option<String>,
+}
+
+#[derive(Clone, Debug, Default)]
+/// Represents a request to update the status of an asset endpoint profile in the ADR Service.
+pub struct UpdateAssetEndpointProfileStatusReq {
+    /// A collection of errors associated with the asset endpiint profile status request.
+    pub errors: Option<Vec<AkriError>>,
+}
+
+impl From<UpdateAssetEndpointProfileStatusReq> for UpdateAssetEndpointProfileStatusRequestPayload {
+    fn from(source: UpdateAssetEndpointProfileStatusReq) -> Self {
+        let errors = source
+            .errors
+            .unwrap_or_default()
+            .into_iter()
+            .map(AkriError::into)
+            .collect();
+
+        UpdateAssetEndpointProfileStatusRequestPayload {
+            asset_endpoint_profile_status_update: AssetEndpointProfileStatus {
+                errors: Some(errors),
+            },
+        }
+    }
+}
+
+/// Request to update the status of an asset in the ADR Service.
+#[derive(Clone, Debug, Default)]
+pub struct UpdateAssetStatusReq {
+    /// The name of the asset whose status is being updated.
+    pub name: String,
+    /// The status of the asset to be updated.
+    pub status: AssetStatusReq,
```

Comment:
What does the Req stand for here?

Reply:
all things ending with req have been renamed